modules/dbparser/patterndb.c

```diff
@@ -205,7 +204,7 @@ _is_action_within_rate_limit(PatternDB *db, PDBProcessParams *process_params)
   PDBRule *rule = process_params->rule;
   PDBAction *action = process_params->action;
   LogMessage *msg = process_params->msg;
-  GString *buffer = process_params->buffer;
+  GString *buffer = g_string_sized_new(256);
```

Comment:
Being this buffer local, it is not needed to steal anymore, I think we do not need to `g_string_steal` the string. For now the stolen string is leaked. (It is not replaced with scratchbuffers in the end)

Reply:
fixed.